client.go

```diff
@@ -23,9 +35,48 @@ func (c client) doRequest(ctx context.Context, in *request, out interface{}) err
 	if err != nil {
 		return err
 	}
-	bytes, err := c.get(ctx, url)
+
+	resp, err := c.get(ctx, url)
 	if err != nil {
 		return err
 	}
-	return json.Unmarshal(bytes, out)
+	defer closeSilently(resp.Body)
+
+	// OSRM returns HTTP status code 200 in case of success or HTTP status code 400 in case of error
+	// In other cases it has to return error
```

Comment:
OSRM returns both codes 200 and 400 in a case with a body. 
In other cases, it returns an unexpected error without a body.

Reply:
updated